portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -68,53 +77,63 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    response.setContentType("text/html; charset=UTF-8");
-
-    if (commented) {
-        response.getWriter().println("<script>alert(\"感谢你的支持！\");</script>");
-        commented = false;
-    }
+    response.setContentType("text/html; charset=utf-8");
 
     for (Entity comments: results.asIterable()) {
         String timestamp = (String) comments.getProperty("timestamp");
         int year = Integer.parseInt(timestamp.substring(0,4));
         int month = Integer.parseInt(timestamp.substring(5,7));
         int day = Integer.parseInt(timestamp.substring(8,10));
-        if (year <= 2020 && month <= 7 && day <= 10) continue;
-        response.getWriter().println(timestamp + "<br></br>");
-        String tmp_email = (String) comments.getProperty("email");
-        int nickname = tmp_email.indexOf('@');
-        if (nickname != -1) { 
-            tmp_email = tmp_email.substring(0, nickname); 
-        }
-        response.getWriter().println(tmp_email + "<br></br>");
-        response.getWriter().println(comments.getProperty("content") + "<br></br>");
+        if (year <= 2020 && month <= 7 && day <= 19) continue;
         String image_url = (String) comments.getProperty("image");
-        if (image_url != null) {
-            response.getWriter().println("<img style=\"height=50px;\" src=\"" + image_url + "\" /><br></br>");
-        }       
+        response.getWriter().println("<article class = \"comment-body\">");
+        response.getWriter().println("<footer class = \"comment-metadata\">");
+        response.getWriter().println("<div class = \"comment-author\">");
+        response.getWriter().println("<img width=\"80\" height=\"80\" src=\"" + image_url + "\" />");
+        String name = (String) comments.getProperty("name");
+        String email = (String) comments.getProperty("email");
+        response.getWriter().println("<b class=\"fn\">" + name + "</b>");
+        response.getWriter().println("<time>" + timestamp + "</time></div></footer>");
+        response.getWriter().println("<div class=\"comment-content\"><p>");
+        response.getWriter().println(comments.getProperty("content") + "</p></div></article>");
+               
     }
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String comment = Jsoup.clean(request.getParameter("text-input"), Whitelist.none());
-      String email;
-      boolean Friend = Boolean.parseBoolean(getParameter(request, "friend", "false"));
+      //String comment = Jsoup.clean(request.getParameter("text-input"), Whitelist.none(), Document.outputSettings().charset("UTF-8"));
+      String pattern = "<.*>.*</.*>";
```

Comment:
Looks like Jsoup.clean still not work for you for UTF-8 charset? It is quite OK to to use string matches. Maybe you can try:

```suggestion
      Document.OutputSettings outputSettings = new Document.OutputSettings().prettyPrint(false).charset("UTF-8");
      String comment = Jsoup.clean(request.getParameter("text-input"), "", Whitelist.simpleText(), outputSettings);
      String pattern = "<.*>.*</.*>";
```